staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -2837,3 +2838,149 @@ func forceRequestWatchProgressSupport(t *testing.T) {
 		t.Fatalf("failed to wait for required %v storage feature to initialize", storage.RequestWatchProgress)
 	}
 }
+
+func TestListIndexer(t *testing.T) {
+	ctx, cacher, terminate := testSetup(t, withSpecNodeNameIndexerFuncs)
+	t.Cleanup(terminate)
+	tests := []struct {
+		name               string
+		requestedNamespace string
+		recursive          bool
+		fieldSelector      fields.Selector
+		indexFields        []string
+		expectIndex        string
+	}{
+		{
+			name:          "request without namespace, without field selector",
+			recursive:     true,
+			fieldSelector: fields.Everything(),
+		},
+		{
+			name:          "request without namespace, field selector with metadata.namespace",
+			recursive:     true,
+			fieldSelector: fields.ParseSelectorOrDie("metadata.namespace=t2-ns1"),
```

Comment:
I'm not a fan that you're encoding the test number inside it.
If someone will decide to add a new testcase in the middle (which might make sense if it's similar to some other one to keep them close to each other), then it requires changing a bunch of others.

Unless I'm missing something, you're not relying on:
1) specific namespace - all namespaces in this test are equivalent
2) specific node - all nodes are equivalent in this test

so I would suggest:
- hardcoding the namespace and use the exact same namespace across all testcases [then you can easily hardcode it in all testcases]
- doing the same for the node

[and you get distinguishing across tests via running them in a separate t.Run() call

Reply:
I copied this approach from watch indexer tests.

Reply:
I know - that doesn't immediately mean we should copy it :)

Reply:
I mean didn't want to be inconsistent with similar tests, assuming that someone else put thought into it.

Reply:
> [and you get distinguishing across tests via running them in a separate t.Run() call

Are you suggesting to add the number during the t.Run() or setup a separate clean etcd every t.Run()?

Reply:
> Are you suggesting to add the number during the t.Run() or setup a separate clean etcd every t.Run()?

The former
Or even better, using exactly the same one in all tests (same for nodeName).

Reply:
Done